app/credentials/concerns/user_provided_services.rb

```diff
@@ -0,0 +1,17 @@
+module UserProvidedService
+  def credentials(service_name)
```

Comment:
Surely the arg here is the attribute, since they're all being pulled from the same service?